lib/parallel_cucumber/runtime.js

```diff
@@ -195,33 +196,94 @@ var Runtime = function(configuration) {
     var tasks = [];
     var taskCount = 0;
 
-    options.featureFilePaths.forEach(function(featureFilePath) {
-      Object.keys(options.profiles).forEach(function(profileName) {
-        var taskIndex = taskCount++;
-        var profile = options.profiles[profileName];
-        // Clone the array using slice()
-        var tags = profile.tags ? profile.tags.slice() : [];
-        var env = profile.env || {};
-
-        env.PARALLEL_CUCUMBER_PROFILE = profileName;
-
-        var task = {
-          taskIndex: taskIndex,
-          profileName: profileName,
-          featureFilePath: featureFilePath,
-          supportCodePaths: options.supportCodePaths,
-          tags: tags,
-          env: env,
-          retryCount: 0
-        };
-
-        tasks.push(task);
+    options.featureFilePaths.forEach(function (featureFilePath) {
+      self._getFeatureFileObjects(featureFilePath).forEach(function (featureFile) {
+        Object.keys(options.profiles).forEach(function (profileName) {
+          var profile = options.profiles[profileName];
+          // Clone the array using slice()
+          var tags = profile.tags ? profile.tags.slice() : [];
+          var env = profile.env || {};
+
+          env.PARALLEL_CUCUMBER_PROFILE = profileName;
+
+          var matchTags = true;
+          tags.forEach(function (tag) {
+            if (matchTags) {
+              if (tag.indexOf('~') === 0) {
+                matchTags = featureFile.tags.indexOf(tag.substring(1)) === -1;
+              }
+              else {
+                matchTags = featureFile.tags.indexOf(tag) > -1;
+              }
+            }
+          });
+          if (matchTags) {
+            var task = {
+              taskIndex: taskCount++,
+              profileName: profileName,
+              featureFilePath: featureFile.path,
+              supportCodePaths: options.supportCodePaths,
+              tags: tags,
+              env: env,
+              retryCount: 0
+            };
+            tasks.push(task);
+          }
+        });
       });
     });
 
     callback(null, tasks);
   };
 
+  /**
```

Comment:
@peter-reperger could you remove the documentation comment from the method, just to maintain the style of the codebase